library/a-star/src/main/java/Main.java

```diff
@@ -74,86 +61,13 @@ public static void main(String[] a) throws IOException, ParseException {
         System.out.println(y2);
 
 
-        try {
-            Spot[][] grid = createGridFromMap("media/h9275.png", startEnd);
-            drawMap(grid);
-            linkNeighbors(grid);
-            Spot walk = algo(grid, x1, y1, x2, y2);
-            drawPath(walk);
-            System.out.println("Done!");
-        } catch (IOException e) {
-            System.out.println("Image does not exist");
-        }
-
-    }
-
-
-    public static Dictionary minmaxXY(JSONObject coords){
-        // The min max default values are set
-        // respectively to the 25x25 array size
-        // generated by the metadata tool
-        long minX = 25;
-        long minY = 25;
-        long maxX = 0;
-        long maxY = 0;
-        for(int i = 0; i<coords.size(); i++){
-            JSONObject xy = (JSONObject) coords.get(String.valueOf(i));
-            long x = (long) xy.get("x");
-            long y = (long) xy.get("y");
-
-            if(x<minX){
-                minX = x;
-            }
-            if(x>maxX){
-                maxX = x;
-            }
-            if(y<minY){
-                minY = y;
-            }
-            if(y>maxY){
-               maxY = y;
-            }
-        }
-
-        Dictionary minmax = new Hashtable();
-        minmax.put("minX", minX*11);
-        minmax.put("maxX", (maxX*11)+11);
-        minmax.put("minY", (minY*11)-11);
-        minmax.put("maxY", maxY*11);
-
-        return minmax;
-    }
-
-    //fixed
-    public static Spot[][] createGridFromMap(String path, Dictionary[] startEnd) throws IOException {
-
-
-        BufferedImage image = ImageIO.read(new File(path));
-
-        int[][] result = getRGBarray(image);
-        boolean[][] bool = getBoolArr(result, -534826);
-
-        Spot[][] grid = new Spot[bool.length][bool[bool.length - 1].length];
-
-        for(int i = 0; i<grid.length; i++){
-            for(int j = 0; j<grid[i].length; j++){
-                    if(i<=Math.toIntExact((long)startEnd[0].get("maxX"))
-                            && i>=Math.toIntExact((long)startEnd[0].get("minX"))
-                            && j<=Math.toIntExact((long)startEnd[0].get("maxY"))
-                            && j>=Math.toIntExact((long)startEnd[0].get("minY"))) {
-                        grid[i][j] = new Spot(i,j,false);
-                    }else if(i<=Math.toIntExact((long)startEnd[1].get("maxX"))
-                                && i>=Math.toIntExact((long)startEnd[1].get("minX"))
-                                && j<=Math.toIntExact((long)startEnd[1].get("maxY"))
-                                && j>=Math.toIntExact((long)startEnd[1].get("minY"))){
-                        grid[i][j] = new Spot(i,j,false);
-                    }else {
-                        grid[i][j] = new Spot(i, j, bool[i][j]);
-                    }
-            }
-        }
-
-        return grid;
+        boolean[][] binGrid = createBinaryGrid("media/h9275.png");
```

Comment:
I think we should add functionality for extending to different buildings/floors as soon as we can. Anything in mind?

Reply:
I am thinking about introducing a builder pattern in order to have more control over the Spot objects being generated. I believe this will allow us, to tackle the multi-floor feature as well as the accessibility, given the metadata.